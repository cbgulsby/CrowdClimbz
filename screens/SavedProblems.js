import React from 'react';
import { 
    StyleSheet, 
    View, 
    Text,
    SafeAreaView,
    Image,
    FlatList,
    TouchableOpacity
} from 'react-native';


const sampleData = [
    {
        id: '1',
        problemName: 'I Hate Mondays',
        gymLocation: 'Birmingham Boulders',
        problemLevel: 2,
    },
    {
        id: '2',
        problemName: 'Going Up',
        gymLocation: 'Birmingham Boulders',
        problemLevel: 0,
    },
    {
        id: '3',
        problemName: 'Going Down',
        gymLocation: 'High Point Birmingham',
        problemLevel: 5,
    },
    {
        id: '4',
        problemName: 'Bottomless Fries',
        gymLocation: 'Birmingham Boulders',
        problemLevel: 3,
    },
    {
        id: '5',
        problemName: 'All Uphill From Here',
        gymLocation: 'Birmingham Boulders',
        problemLevel: 4,
    },
    {
        id: '6',
        problemName: 'Angry Monkeys',
        gymLocation: 'High Point Birmingham',
        problemLevel: 4,
    },
    {
        id: '7',
        problemName: 'Jeon Jungkook',
        gymLocation: 'B.bloc Climbing Gangnam',
        problemLevel: 9,
    },
    {
        id: '8',
        problemName: 'Seagulls Stop It',
        gymLocation: 'Dagobah',
        problemLevel: 2,
    },
    {
        id: '9',
        problemName: 'Space Mountain',
        gymLocation: 'Witt Center',
        problemLevel: 1,
    },
]


export default function SavedProblems(){
    return(
        <SafeAreaView style={styles.container}>
            <Text> Saved Problems </Text>

            <FlatList
                data={sampleData}
                renderItem={({item}) => <Text />} 
                keyExtractor={item => item.id} 
            />
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        paddingTop: 24,
        backgroundColor: '#4fb9ff',
        flex: 1
    }
});