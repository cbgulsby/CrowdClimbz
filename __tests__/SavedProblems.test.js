import React from 'react';
import renderer from 'react-test-renderer';

import SavedProblems from '../screens/SavedProblems';

// it('works', () => {
//     expect(1).toBe(1);
//   });


test('renders correctly', () => {
  const tree = renderer.create(<SavedProblems />).toJSON();
  expect(tree).toMatchSnapshot();
});