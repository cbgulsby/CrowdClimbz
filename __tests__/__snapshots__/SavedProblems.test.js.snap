// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#4fb9ff",
      "flex": 1,
      "paddingTop": 24,
    }
  }
>
  <Text>
     Saved Problems 
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "gymLocation": "Birmingham Boulders",
          "id": "1",
          "problemLevel": 2,
          "problemName": "I Hate Mondays",
        },
        Object {
          "gymLocation": "Birmingham Boulders",
          "id": "2",
          "problemLevel": 0,
          "problemName": "Going Up",
        },
        Object {
          "gymLocation": "High Point Birmingham",
          "id": "3",
          "problemLevel": 5,
          "problemName": "Going Down",
        },
        Object {
          "gymLocation": "Birmingham Boulders",
          "id": "4",
          "problemLevel": 3,
          "problemName": "Bottomless Fries",
        },
        Object {
          "gymLocation": "Birmingham Boulders",
          "id": "5",
          "problemLevel": 4,
          "problemName": "All Uphill From Here",
        },
        Object {
          "gymLocation": "High Point Birmingham",
          "id": "6",
          "problemLevel": 4,
          "problemName": "Angry Monkeys",
        },
        Object {
          "gymLocation": "B.bloc Climbing Gangnam",
          "id": "7",
          "problemLevel": 9,
          "problemName": "Jeon Jungkook",
        },
        Object {
          "gymLocation": "Dagobah",
          "id": "8",
          "problemLevel": 2,
          "problemName": "Seagulls Stop It",
        },
        Object {
          "gymLocation": "Witt Center",
          "id": "9",
          "problemLevel": 1,
          "problemName": "Space Mountain",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                I Hate Mondays
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Birmingham Boulders
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              2
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Going Up
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Birmingham Boulders
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              0
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Going Down
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                High Point Birmingham
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              5
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Bottomless Fries
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Birmingham Boulders
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              3
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                All Uphill From Here
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Birmingham Boulders
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              4
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Angry Monkeys
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                High Point Birmingham
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              4
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Jeon Jungkook
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                B.bloc Climbing Gangnam
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              9
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Seagulls Stop It
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Dagobah
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              2
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2576f7",
              "borderColor": "black",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 100,
              "margin": 3,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-end",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  }
                }
              >
                Space Mountain
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "flex-start",
                  "paddingLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Witt Center
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 40,
                }
              }
            >
              V
              1
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
